app/scripts/lib/setupSentry.js

```diff
@@ -594,15 +588,72 @@ function addDebugListeners() {
   log('Added debug listeners');
 }
 
+export function fetchWithSentryInstrumentation(fetchFunction) {
+  return async (inputUrl, opts) => {
+    const url =
+      typeof inputUrl === 'string'
+        ? inputUrl
+        : inputUrl.toString() || String(inputUrl);
+    const { method = 'GET' } = opts ?? {};
+
+    const response = await fetchFunction(url, {
+      method,
+      ...opts,
+    });
+
+    // Do not create spans for outgoing requests to a 'sentry.io' domain.
+    if (url.match(/^https?:\/\/([\w\d.@-]+\.)?sentry\.io(\/|$)/u)) {
+      return response;
+    }
+
+    try {
+      Sentry.startSpan(
+        { op: 'http.client', name: `${method} ${url}` },
+        (span) => {
+          const parsedURL = new URL(url, globalThis.location?.origin);
+
+          span.setAttributes({
+            'http.request.method': method,
+            'server.address': parsedURL.hostname,
+            'server.port': parsedURL.port || undefined,
+            'http.response.status_code': response.status,
+            'http.response_content_length': Number(
+              response.headers?.get('content-length'),
+            ),
```

Comment:
Since not all requests will have `content-length` header, this code will result in `'http.response_content_length': NaN`. I think it should be set to `undefined` (or this property just not defined at all).